From 98823cdec5eeafae9b7ed17be4e909314e924111 Mon Sep 17 00:00:00 2001
From: Zhipeng Xie <xiezhipeng1@huawei.com>
Date: Thu, 27 Feb 2020 15:36:55 -0500
Subject: [PATCH 15/37] create-diff-object: fix .orc_unwind_ip error

error: .orc_unwind_ip section header details
differ from .orc_unwind_ip

Don't correlate .orc_unwind sections and symbols

Signed-off-by: Zhipeng Xie <xiezhipeng1@huawei.com>
---
 kpatch-build/create-diff-object.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/kpatch-build/create-diff-object.c b/kpatch-build/create-diff-object.c
index 9b966e8..9c70fee 100644
--- a/kpatch-build/create-diff-object.c
+++ b/kpatch-build/create-diff-object.c
@@ -1061,6 +1061,9 @@ static void kpatch_correlate_sections(struct list_head *seclist_orig,
 					continue;
 			}
 
+			if (strstr(sec_orig->name, ".orc_unwind"))
+				continue;
+
 			kpatch_correlate_section(sec_orig, sec_patched);
 			break;
 		}
@@ -1098,6 +1101,9 @@ static void kpatch_correlate_symbols(struct list_head *symlist_orig,
 			    sym_orig->sec->twin != sym_patched->sec)
 				continue;
 
+			if (strstr(sym_orig->name, ".orc_unwind"))
+				continue;
+
 			kpatch_correlate_symbol(sym_orig, sym_patched);
 			break;
 		}
-- 
2.33.0

